course/include/isometry.h

```diff
@@ -96,5 +96,93 @@ inline std::ostream& operator<<(std::ostream& os, const Vector3& vector) {
               << ", z: " << std::to_string(static_cast<int>(vector.z())) << ")";
 }
 
+struct Rows {
+    Rows() : r1_(), r2_(), r3_() {};
+    Rows(const Vector3 &r1, const Vector3 &r2, const Vector3 &r3) : r1_(r1), r2_(r2), r3_(r3) {}; 
+    Vector3 r1_,r2_,r3_;
+};
+
+
+class Matrix3 {
+
+    public:
+        Matrix3() : m_(new Rows{}) {};
+
+        Matrix3(const Vector3 &r1, const Vector3 &r2, const Vector3 &r3) : m_(new Rows{r1, r2, r3}) {};
+        Matrix3(const std::initializer_list<double>& values);
+        ~Matrix3() {
+            // deallocate
+            delete m_;  
+        };
+
+        // copy constructor
+        Matrix3(const Matrix3& other) : m_(new Rows{*(other.m_)}) {};
```

Comment:
Remember to deallocate the memory that was previously in `m_`, otherwise you're leaking memory.

Reply:
Theoretically, the m_ is a null pointer because I am creating the object if I don't understand correctly, and anyway in the destructor, I am releasing the memory. Maybe I am missing something.

Reply:
No, you're right. I was thinking of the assignment operation. My mistake, sorry!